staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -0,0 +1,264 @@
+// today is not that day.
+type FairQueuingSystem interface {
+
+	// The following methods modify or query the configuration
+
+	SetConcurrencyLimit(int)
+	SetDesiredNumQueues(int)
+	GetDesiredNumQueues() int
+	GetActualNumQueues() int
+	SetQueueLengthLimit(int)
+	SetRequestWaitLimit(time.Duration)
+
+	// DoOnEmpty records a function to be called when all the queues
+	// have no requests waiting nor executing.  The filter uses this
+	// for a priority level that has become undesired, setting a
+	// callback that will actually remove the priority level by
+	// establishing a new RMState whose priorityLevelStates map lacks
+	// that entry.  If the priority level becomes desired again before
+	// this is called, the filter cancels by calling `DoOnEmpty(nil)`.
+	// The callback is invoked with no mutexes locked.
+	DoOnEmpty(func())
+
+	// Wait, in the happy case, shuffle shards the given request into
+	// a queue and eventually dispatches the request from that queue.
+	// Dispatching means to return with `execute==true`.  In the
+	// unhappy cases the request is eventually rejected, which means
+	// to return with `execute=false`.  In the happy case the caller
+	// is required to invoke the returned `afterExecution` after the
+	// request is done executing.  The hash value and hand size are
+	// used to do the shuffle sharding.
+	Wait(hashValue uint64, handSize int32) (execute bool, afterExecution func())
+}
+
+// RMState is the variable state that this filter is working with at a
+// given point in time.
+type RMState struct {
+	// flowSchemas holds the flow schema objects, sorted by increasing
+	// numerical (decreasing logical) matching precedence
+	flowSchemas FlowSchemaSeq
+
+	// priorityLevelStates maps the PriorityLevelConfiguration object
+	// name to the state for that level
+	priorityLevelStates map[string]*PriorityLevelState
+}
+
+// FlowSchemaSeq holds sorted set of pointers to FlowSchema objects.
+// FLowSchemaSeq implements `sort.Interface` (TODO: implement this).
+type FlowSchemaSeq []*rmtypesv1a1.FlowSchema
+
+// PriorityLevelState holds the state specific to a priority level.
+// golint requires that I write something here,
+// even if I can not think of something better than a tautology.
+type PriorityLevelState struct {
+	// config holds the configuration after defaulting logic has been applied
+	config rmtypesv1a1.PriorityLevelConfigurationSpec
+
+	// concurrencyLimit is the limit on number executing
+	concurrencyLimit int
+
+	fqs FairQueuingSystem
+}
+
+// requestManagement holds all the state and infrastructure of this
+// filter
+type requestManagement struct {
+	fairQueuingFactory FairQueuingFactory
+
+	// configQueue holds TypedConfigObjectReference values, identifying
+	// config objects that need to be processed
+	configQueue workqueue.RateLimitingInterface
+
+	// numConfigWorkers is the configured number of workers for the config queue
+	numConfigWorkers int
```

Comment:
i think it's a bit overkilling.. it's a good idea to load the config(i.e. PL)  changes in an asynchronous way but do we really need a configurable number of workers in the system? one worker will be sufficient i presume because we don't expect frequent too many PL and PL changes

Reply:
Yes, actually I think it is positively good to have just one worker --- that eliminates the need for synchronization among workers.  I will update.